### PR TITLE
chore: remove urgent c2 (FPLA-3093)

### PR DESCRIPTION
**Pull request by toireasam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-3080

### Change description ###
Revert what was done in previous migration as user added another application. this time checking id's to not cause this issue again

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 1]
```diff
@@ -28,7 +35,9 @@
 @Slf4j
 public class MigrateCaseController extends CallbackController {
     private static final String MIGRATION_ID_KEY = "migrationId";
-    private final CaseSubmissionService caseSubmissionService;
+    private static final String ID_FIRST_BUNDLE = "fbf05208-f5dd-4942-b735-9aa226d73a2e";
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:38]: would it be better to store them as uuids instead, you are doing all the comparisons later by converting the uuids to strings or the other way round

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 2]
```diff
@@ -70,22 +79,106 @@ private void run3037(CaseDetails caseDetails) {
...
 
-        additionalApplicationsBundle.remove(0);
+        Element<AdditionalApplicationsBundle> secondBundle = additionalApplicationsBundle.get(1);
 
-        Map<String, Object> data = caseDetails.getData();
+        if (secondBundle.getId().equals(UUID.fromString(ID_SECOND_BUNDLE))) {
+            swapC2WithSupportingDocument(secondBundle.getValue(), additionalApplicationsBundle);
+        } else {
+            throw new IllegalStateException(String
+                .format("Migration failed on case %s: Expected " + ID_SECOND_BUNDLE + " but got %s",
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:116]: you are using string.format but then doing concatenation, do one or the other

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 3]
```diff
@@ -70,22 +79,106 @@ private void run3037(CaseDetails caseDetails) {
...
+                    .build())
+                .build());
+
+            additionalApplicationsBundle.set(0, firstBundle);
+
+        } else {
             throw new IllegalStateException(String
-                .format("Migration failed on case %s: Case has %s additional applications",
-                    caseData.getFamilyManCaseNumber(), additionalApplicationsBundle.size()));
+                .format("Migration failed on case %s: Expected " + ID_FIRST_BUNDLE + " but got %s",
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:106]: you are using string.format but then doing concatenation, do one or the other

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 4]
```diff
@@ -70,22 +79,105 @@ private void run3037(CaseDetails caseDetails) {
...
+                .document(supportingEvidenceBundle.get().getValue().getDocument())
+                .build());
+        }
+
+        c2DocumentBundle.getSupportingEvidenceBundle()
+            .removeIf(bundle -> bundle.getId()
+                .equals(SUPPORTING_EVIDENCE_ID)
+            );
+
+        additionalApplicationsBundle.set(1, element(application));
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:181]: vv minor but I think the should be the responsibility of the calling method

**Review by lewisbirks — APPROVED** [review]
i think it is fine but it might be good to get another set of eyes on it just in case as we don't want to mess up again

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 5]
```diff
@@ -70,22 +79,105 @@ private void run3037(CaseDetails caseDetails) {
...
+    private void swapC2WithSupportingDocument(AdditionalApplicationsBundle application,
+              List<Element<AdditionalApplicationsBundle>> additionalApplicationsBundle) {
+        C2DocumentBundle c2DocumentBundle = application.getC2DocumentBundle();
+        Optional<Element<SupportingEvidenceBundle>> supportingEvidenceBundle = c2DocumentBundle
+            .getSupportingEvidenceBundle()
+            .stream()
+            .filter(bundle -> bundle.getId().equals(SUPPORTING_EVIDENCE_ID))
+            .findFirst();
+
+        if (supportingEvidenceBundle.isPresent()) {
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:170]: Is there a possibility that it won't be present?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 6]
```diff
@@ -70,22 +79,105 @@ private void run3037(CaseDetails caseDetails) {
...
 
-    private void run3093(CaseDetails caseDetails) {
+    private void run3080(CaseDetails caseDetails) {
         CaseData caseData = getCaseData(caseDetails);
 
         List<Element<AdditionalApplicationsBundle>> additionalApplicationsBundle = caseData
             .getAdditionalApplicationsBundle();
 
-        if (caseData.getAdditionalApplicationsBundle().size() < 1) {
+        Element<AdditionalApplicationsBundle> firstBundle = additionalApplicationsBundle.get(0);
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:88]: this one still may fail if new application uploaded meanwhile, but thats ok, migration will fail

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 7]
```diff
@@ -70,22 +79,105 @@ private void run3037(CaseDetails caseDetails) {
...
+                .format("Migration failed on case %s: Expected %s but got %s",
+                    caseData.getFamilyManCaseNumber(), SECOND_BUNDLE_ID, secondBundle.getId()));
+        }
 
+        Map<String, Object> data = caseDetails.getData();
         data.put("additionalApplicationsBundle", additionalApplicationsBundle);
     }
+
+    private List<Element<SupportingEvidenceBundle>> buildSupportingEvidenceBundle() {
+        SupportingEvidenceBundle firstBundle = SupportingEvidenceBundle.builder()
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:125]: the risk is that user could remove supporting doc via manage doc event, so restoring will recreate wrong state, but, we can accept risk

**Review by tomaszpowroznik — APPROVED** [review]

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java (hunk 5 above)
Comment (toireasam) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:170]: no guess not - will remove :)

**Review by lewisbirks — APPROVED** [review]

**Pull request merged**